neuralforecast/models/timellm.py

```diff
@@ -0,0 +1,497 @@
+        hist_exog_list=None,
+        futr_exog_list=None,
+        loss=MAE(),
+        valid_loss=None,
+        learning_rate: float = 1e-4,
+        max_steps: int = 5,
+        val_check_steps: int = 100,
+        batch_size: int = 32,
+        valid_batch_size: Optional[int] = None,
+        windows_batch_size: int = 1024,
+        inference_windows_batch_size: int = 1024,
+        start_padding_enabled: bool = False,
+        step_size: int = 1,
+        num_lr_decays: int = 0,
+        early_stop_patience_steps: int = -1,
+        scaler_type: str = "identity",
+        num_workers_loader: int = 0,
+        drop_last_loader: bool = False,
+        random_seed: int = 1,
+        **trainer_kwargs,
+    ):
+        super(TimeLLM, self).__init__(
+            h=h,
+            input_size=input_size,
+            hist_exog_list=hist_exog_list,
+            stat_exog_list=stat_exog_list,
+            futr_exog_list=futr_exog_list,
+            loss=loss,
+            valid_loss=valid_loss,
+            max_steps=max_steps,
+            learning_rate=learning_rate,
+            num_lr_decays=num_lr_decays,
+            early_stop_patience_steps=early_stop_patience_steps,
+            val_check_steps=val_check_steps,
+            batch_size=batch_size,
+            valid_batch_size=valid_batch_size,
+            windows_batch_size=windows_batch_size,
+            inference_windows_batch_size=inference_windows_batch_size,
+            start_padding_enabled=start_padding_enabled,
+            step_size=step_size,
+            scaler_type=scaler_type,
+            num_workers_loader=num_workers_loader,
+            drop_last_loader=drop_last_loader,
+            random_seed=random_seed,
+            **trainer_kwargs,
+        )
+
+        # Asserts
+        if stat_exog_list is not None:
+            raise Exception("TimeLLM does not support static exogenous variables")
+        if futr_exog_list is not None:
+            raise Exception("TimeLLM does not support future exogenous variables")
+        if hist_exog_list is not None:
+            raise Exception("TimeLLM does not support historical exogenous variables")
+
+        # Architecture
+        self.patch_len = patch_len
+        self.stride = stride
+        self.d_ff = d_ff
+        self.top_k = top_k
+        self.d_llm = d_llm
+        self.d_model = d_model
+        self.dropout = dropout
+        self.n_heads = n_heads
+        self.enc_in = enc_in
+        self.dec_in = dec_in
+
+        self.llm_config = llm_config
+        self.llm = llm
+        self.llm_tokenizer = llm_tokenizer
+
+        if self.llm_config is None:
+            self.llm_confg = GPT2Config.from_pretrained("openai-community/gpt2")
+        if self.llm is None:
+            self.llm = GPT2Model.from_pretrained(
+                "openai-community/gpt2", config=self.llm_confg
+            )
+        if self.llm_tokenizer is None:
+            self.llm_tokenizer = GPT2Tokenizer.from_pretrained("openai-community/gpt2")
```

Comment:
i think it might be better if we use `GPT2` when `llm is None`, I'm thinking for example, what happens if `llm_tokenizer is not None` but for some reason `llm is None`? also, we should add an exception based on `IS_TRANSFORMERS_INSTALLED` in case that the default params are being used, something like:

```python
if self.llm is None:
      if not IS_TRANSFORMERS_INSTALLED:
             raise ImportError("please install `transformers` to use default llm")
      print("Using GPT2 model as default and ignoring `llm_config` and `llm_tokenizer`")
      self.llm = GPT2...
      self.llm_config = GPT2...
      self.llm_tokenizer = GPT2...
```